external/loaders/loaders/mappers/_hybrid.py

```diff
@@ -313,52 +321,120 @@ def _add_fine_res_inputs(fine: xarray.Dataset, input_feature_url: Optional[str]
     return merged.astype(numpy.float32)
 
 
-@mapper_functions.register
-def open_fine_resolution(
-    fine_url: str = "",
-    input_feature_url: Optional[str] = None,
-    include_temperature_nudging: bool = False,
-) -> GeoMapper:
-    """
-    Open the fine-res mapper optionally using state from another run.
+def open_fine_resolution_additional_source(
+    fine_url: str,
+    input_feature_url: str,
+    additional_source_url: str,
+    include_temperature_nudging: bool,
+) -> xarray.Dataset:
 
-    Args:
-        fine_url: url where coarsened fine resolution data is stored
-        input_feature_url: url to fv3gfs_run to use for state inputs. If not provided,
-            the state of the fine-res data will be used as input.
-        include_temperature_nudging: whether to include fine-res nudging in Q1
+    fine = _compute_fine_res_portion(fine_url, include_temperature_nudging)
+    additional_source = open_zarr_maybe_consolidated(additional_source_url)
 
-    Returns:
-        a mapper
-    """
-    return XarrayMapper(
-        open_fine_resolution_dataset(
-            fine_url=fine_url,
-            input_feature_url=input_feature_url,
-            include_temperature_nudging=include_temperature_nudging,
-        )
+    fine["Q1"] = fine.Q1 + additional_source.Q1
```

Comment:
Not sure if the "additional source" will always have Q1, Q2 (obviously some renaming of the nudging tendencies is needed), could split this into separate options for the nudging tendency and dynamics difference.

Reply:
Maybe this function should take an argument for the names of the additional sources? Then these could be hard-coded in the function one level up. E.g. for nudging tendencies it would be

`{"Q1": "air_temperature_tendency_due_to_nudging", "Q2": "specific_humidity_tendency_due_to_nudging"}`

Reply:
Do we need an "additional source" configuration? I could see this getting pretty confusing and makes the config less declarative because it starts to describe behavior (i.e. addition).

Reply:
OK it sounds like the preference then is to just split this into two declarative approaches that are self-evident (eg `apparent_sources_with_nudging` and `apparent_sources_with_dynamics_differences`)